interpreter/interpreter.go

```diff
@@ -230,12 +230,27 @@ func (itrp *Interpreter) mutateFunctionScope(f function) (function, error) {
 	// copy the scope so we can safely mutate it
 	f.scope = f.scope.Copy()
 	copyPackages(f.scope)
+	mutatedPkg := false
 	for _, mut := range itrp.modifiedOptions {
+		// Check if the function is defined in the scope of package that was mutated
```

Comment:
Can you explain this new addition? I'm not quite sure I follow.

Reply:
Yeah, given this is a builtin package

```
package foo

option bar = 1

do = () => bar
```

Then I do this in main:

```
package main

import "foo"
option foo.bar = 2

foo.do() // should return 2
```

Before the change it would return `1`. This was because the function `do` would close over its package scope. Calling `SetOption` on the `foo` package later had need effect on the package scope from within the `foo` package because the package scope is implemented by copying the package as an object.

The fix is to check if the scope of the function is of a package that had one of its options mutated. If so we need to do set the option on the package, and we need to recreate the package scope from the modified package.